petab/visualize/helper_functions.py

```diff
@@ -434,44 +445,44 @@ def get_data_to_plot(vis_spec: pd.DataFrame,
 
     for var_cond_id in condition_ids:
         # get boolean vector which fulfill the requirements
-        vec_bool_meas = ((m_data[col_id] == var_cond_id) & (m_data['datasetId']
+        vec_bool_meas = ((m_data[col_id] == var_cond_id) & (m_data[DATASET_ID]
                          == vis_spec.datasetId[i_visu_spec]))
         # get indices of rows with True values of vec_bool_meas
         ind_meas = [i_visu_spec for i_visu_spec,
                     x in enumerate(vec_bool_meas) if x]
 
         # check that all entries for all columns-conditions are the same:
         # check correct observable
-        bool_observable = (m_data.observableParameters[ind_meas[0]] ==
-                           m_data.observableParameters)
+        bool_observable = (m_data[OBSERVABLE_PARAMETERS][ind_meas[0]] ==
+                           m_data[OBSERVABLE_PARAMETERS])
         # special handling, if column in m_data.observableParameters is empty
-        if (type(m_data.observableParameters[ind_meas[0]]) == np.float64) \
+        if (type(m_data[OBSERVABLE_PARAMETERS][ind_meas[0]]) == np.float64) \
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Using type() instead of isinstance() for a typecheck.](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=4881219)